warp/aggregator/aggregator.go

```diff
@@ -36,16 +62,133 @@ func (a *Aggregator) AggregateSignatures(ctx context.Context, unsignedMessage *a
 	if err != nil {
 		return nil, err
 	}
-	job := newSignatureAggregationJob(
-		a.client,
-		pChainHeight,
-		a.subnetID,
-		quorumNum,
-		quorumNum,
-		params.WarpQuorumDenominator,
-		a.state,
-		unsignedMessage,
+
+	log.Info("Fetching signature",
+		"a.subnetID", a.subnetID,
+		"height", pChainHeight,
 	)
+	validators, totalWeight, err := avalancheWarp.GetCanonicalValidatorSet(ctx, a.state, pChainHeight, a.subnetID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get validator set: %w", err)
+	}
+	if len(validators) == 0 {
+		return nil, fmt.Errorf("%w (SubnetID: %s, Height: %d)", errNoValidators, a.subnetID, pChainHeight)
+	}
+
+	return aggregateSignatures(ctx, a.client, unsignedMessage, validators, quorumNum, totalWeight)
+}
+
+func aggregateSignatures(
```

Comment:
I thought it made sense to split `AggregateSignatures` into the part that gets the required state from `a` and the part that deals with actually acquiring and aggregating the signatures but I could also just combine these back into 1 function.

Reply:
I have no strong opinion but seems we can combine those two